order/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+    pub order_type: OrderType,
+    pub fill_type: FillType,
+    pub nonce: u32,
+    pub valid_to: u32,
+    pub signature: Signature,
+}
+
+/// An order that is returned when querying the orderbook.
+///
+/// Contains extra fields thats are populated by the orderbook.
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Full {
+    pub creation_time: DateTime<Utc>,
+    #[serde(with = "h160_hex")]
+    pub owner: H160,
+    #[serde(flatten)]
+    pub user_provided: UserProvided,
+}
+
+impl Serialize for Signature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut bytes = [0u8; 65 * 2];
+        // Can only fail if the buffer size does not match but we know it is correct.
+        hex::encode_to_slice([self.v], &mut bytes[..2]).unwrap();
+        hex::encode_to_slice(self.r, &mut bytes[2..66]).unwrap();
+        hex::encode_to_slice(self.s, &mut bytes[66..]).unwrap();
+        // Hex encoding is always valid utf8.
+        let str = std::str::from_utf8(&bytes).unwrap();
+        serializer.serialize_str(str)
+    }
+}
+
+impl<'de> Deserialize<'de> for Signature {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct Visitor {}
+        impl<'de> de::Visitor<'de> for Visitor {
+            type Value = Signature;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                write!(formatter, "the 65 signature bytes as a hex encoded string")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                let mut bytes = [0u8; 65];
+                hex::decode_to_slice(s, &mut bytes).map_err(|err| {
+                    de::Error::custom(format!("failed to decode {:?} as hex: {}", s, err))
+                })?;
+                // Can't directly assign here because of type system limitations.
+                let mut r = [0u8; 32];
+                r.copy_from_slice(&bytes[1..33]);
+                let mut s = [0u8; 32];
+                s.copy_from_slice(&bytes[33..]);
+                Ok(Signature { v: bytes[0], r, s })
+            }
+        }
+
+        deserializer.deserialize_str(Visitor {})
+    }
+}
+
+mod h160_hex {
+    use primitive_types::H160;
+    use serde::{de, Deserializer, Serializer};
+    use std::fmt;
+
+    pub fn serialize<S>(value: &H160, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut bytes = [0u8; 20 * 2];
+        // Can only fail if the buffer size does not match but we know it is correct.
+        hex::encode_to_slice(value, &mut bytes).unwrap();
+        // Hex encoding is always valid utf8.
+        let s = std::str::from_utf8(&bytes).unwrap();
+        serializer.serialize_str(s)
```

Comment:
nit: it might be slightly simpler to use the [LowerHex implementation of H160](https://tomusdrw.github.io/rust-web3/web3/types/struct.H160.html#impl-LowerHex)

```rust
serializer.serialize_str(format!("{:x}", value))
```

Reply:
We want `hex` anyway for the u8 slices because those implement LowerHex and then I like using it for the address too.  The current implementation also doesn't allocate (but same can be achieved with `write!` instead of `format!`).